core/src/bank_forks.rs

```diff
@@ -170,11 +175,11 @@ impl BankForks {
                 .map(|p| p.slot())
                 .rev()
                 .skip(1);
-            self.slots_since_snapshot.extend(new_rooted_path);
-            self.slots_since_snapshot.push(root);
-            if self.slots_since_snapshot.len() > MAX_CACHE_ENTRIES {
-                let num_to_remove = self.slots_since_snapshot.len() - MAX_CACHE_ENTRIES;
-                self.slots_since_snapshot.drain(0..num_to_remove);
+            self.slots_to_snapshot.extend(new_rooted_path);
```

Comment:
b/c we call `root_bank.squash()` right below this which updates the roots in the status cache, I think we can avoid tracking `self. slots_to_snapshot` and just grab the MAX_CACHE_ENTRIES roots directly from the status cache.  

Something like `root_bank.src.status_cache.read().unwrap().roots().clone();` after we call `root_bank.squash()` below

Reply:
hm yeah I think you're right. good catch. I'll try to update to that.